src/traffic_control/add_filter.rs

```diff
@@ -116,6 +116,22 @@ impl TrafficFilterNewRequest {
         self
     }
 
+    /// The matchall filter matches every packet and applies actions, etc
+    pub fn matchall(mut self, data: Vec<tc::matchall::Nla>) -> Self {
+        assert!(!self
```

Comment:
Instead of `assert` which abort the whole program, please use `Result<Self, Error::InvalidNla>`